standalone/src/main/java/io/delta/standalone/storage/LogStore.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright (2021) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.delta.standalone.storage;
+
+import io.delta.standalone.data.CloseableIterator;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileStatus;
+import org.apache.hadoop.fs.Path;
+
+import java.io.FileNotFoundException;
+import java.nio.file.FileAlreadyExistsException;
+import java.util.Iterator;
+
+/**
+ * :: DeveloperApi ::
+ *
+ * General interface for all critical file system operations required to read and write the
+ * Delta logs. The correctness is predicated on the atomicity and durability guarantees of
+ * the implementation of this interface. Specifically,
+ *
+ * 1. Atomic visibility of files: If isPartialWriteVisible is false, any file written through
+ * this store must be made visible atomically. In other words, this should not generate partial
+ * files.
+ *
+ * 2. Mutual exclusion: Only one writer must be able to create (or rename) a file at the final
+ *    destination.
+ *
+ * 3. Consistent listing: Once a file has been written in a directory, all future listings for
+ *    that directory must return that file.
+ *
+ * All subclasses of this interface is required to have a constructor that takes Configuration
+ * as a single parameter. This constructor is used to dynamically create the LogStore.
+ *
+ * LogStore and its implementations are not meant for direct access but for configuration based
+ * on storage system. See [[https://docs.delta.io/latest/delta-storage.html]] for details.
+ *
+ * @since 1.0.0
```

Comment:
We need to make a cleanup after this. The version is not 1.0.0 in delta-standalone.